Updating model card for wav2vec2 (#38956)
huggingface/transformers: docs/source/en/model_doc/wav2vec2.md

">
        <img alt="FlashAttention" src="https://img.shields.io/badge/%E2%9A%A1%EF%B8%8E%20FlashAttention-eae0c8?style=flat">
        <img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
    </div>
</div>

# Wav2Vec2

> [Wav2Vec2](https://huggingface.co/papers/2006.11477) is a self-supervised learning framework for speech representations that masks speech input in the latent space and solves a contrastive task over quantized latent representations. It's like having a speech expert that learns the patterns of human speech from raw audio alone, then fine-tunes on transcribed speech to achieve remarkable accuracy.

Reviewer: ```suggestion
[Wav2Vec2](https://huggingface.co/papers/2006.11477) is a self-supervised learning framework that makes pretraining audio models easier and more efficient. It is pretrained on large amounts of unlabeled data where certain chunks of audio are masked and the model must predict it. The model is fine-tuned on a much smaller labeled dataset and demonstrates it can still achieve competitive results. 
```

> You can find all the original [Wav2Vec2](https://huggingface.co/models?search=wav2vec2) checkpoints on the Hugging Face Hub.

Reviewer: ```suggestion
You can find all the original Wav2Vec2 checkpoints in the [Wav2Vec2.0](https://huggingface.co/collections/facebook/wav2vec-20-651e865258e3dee2586c89f5) collection.
```

> > [!TIP]
> > Click on the Wav2Vec2 models in the right sidebar for more examples of how to apply Wav2Vec2 to different speech recognition and audio classification tasks.

Reviewer: ```suggestion
> [!TIP]
> This model was contributed by [patrickvonplaten](https://huggingface.co/patrickvonplaten).
>
> Click on the Wav2Vec2 models in the right sidebar for more examples of how to apply Wav2Vec2 to different speech recognition and audio classification tasks.
```

The example below demonstrates how to perform automatic speech recognition using the [`Pipeline`] or the [`AutoModel`] class.

<hfoptions id="usage">
<hfoption id="Pipeline">

```python

> from transformers import pipeline

Reviewer: ```py
import torch
from transformers import pipeline

pipeline = pipeline("automatic-speech-recognition", model="facebook/wav2vec2-base-960h", device=0, torch_dtype="auto")
pipeline("https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/1.flac")
```

# Automatic Speech Recognition
asr = pipeline("automatic-speech-recognition", model="facebook/wav2vec2-base-960h")
transcription = asr("path/to/audio.wav")

# Audio Classification
classifier = pipeline("audio-classification", model="facebook/wav2vec2-base")
result = classifier("path/to/audio.wav")
```

</hfoption>

> <hfoption id="AutoModel">

Reviewer: ```py
import torch
from datasets import load_dataset
from transformers import AutoProcessor, AutoModelForCTC

dataset = load_dataset("hf-internal-testing/librispeech_asr_demo", "clean", split="validation")
dataset = dataset.sort("id")
sampling_rate = dataset.features["audio"].sampling_rate

processor = AutoProcessor.from_pretrained("facebook/wav2vec2-base-960h")
model = AutoModelForCTC.from_pretrained("facebook/wav2vec2-base-960h", torch_dtype="auto", device_map="auto", attn_implementation="sdpa")

inputs = processor(dataset[0]["audio"]["array"], sampling_rate=sampling_rate, return_tensors="pt").to("cuda")

with torch.no_grad():
    logits = model(**inputs).logits
predicted_ids = torch.argmax(logits, dim=-1)

transcription = processor.batch_decode(predicted_ids)
transcription[0]
```

```python
from transformers import AutoProcessor, AutoModelForCTC
import torch

# Load model and processor
processor = AutoProcessor.from_pretrained("facebook/wav2vec2-base-960h")
model = AutoModelForCTC.from_pretrained("facebook/wav2vec2-base-960h")

# Prepare audio input (assuming you have audio data)
audio_input = load_audio("path/to/audio.wav")
inputs = processor(audio_input, sampling_rate=16000, return_tensors="pt", padding=True)

Generate logits
with torch.no_grad():
    logits = model(**inputs).logits

Decode
predicted_ids = torch.argmax(logits, dim=-1)
transcription = processor.batch_decode(predicted_ids)
```

</hfoption>
</hfoptions>

> Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.

Reviewer: No need to include a quantization example since the model isn't that large

The example below uses [bitsandbytes quantization](https://huggingface.co/docs/transformers/main/en/main_classes/quantization) to quantize the weights to 8-bit:

```python
from transformers import Wav2Vec2ForCTC
import torch

model = Wav2Vec2ForCTC.from_pretrained(
    "facebook/wav2vec2-base-960h",
    load_in_8bit=True,
    device_map="auto"
)
```

> ## Notes

Reviewer: You can remove all the notes except for:

```md
- Wav2Vec2 is a speech model that accepts a float array corresponding to the raw waveform of the speech signal.
- If you have a `head_mask`, you must use the eager attention implementation for it to work.

   ```py
   from transformers import AutoModelForCTC

   model = AutoModelForCTC.from_pretrained("facebook/wav2vec2-base-960h", attn_implementation="eager")
   ```

### Batch Decoding Example
```python
from multiprocessing import get_context
from transformers import AutoProcessor, AutoModelForCTC
from datasets import load_dataset
import torch

model = AutoModelForCTC.from_pretrained("patrickvonplaten/wav2vec2-base-100h-with-lm").to("cuda")
processor = AutoProcessor.from_pretrained("patrickvonplaten/wav2vec2-base-100h-with-lm")

# Load dataset
dataset = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")

def map_to_pred(batch, pool):
    inputs = processor(batch["speech"], sampling_rate=16_000, padding=True, return_tensors="pt")
    inputs = {k: v.to("cuda") for k, v in inputs.items()}
    
    with torch.no_grad():
        logits = model(**inputs).logits
    
    transcription = processor.batch_decode(logits.cpu().numpy(), pool).text
    batch["transcription"] = transcription
    return batch

# Use managed pool for batch decoding
with get_context("fork").Pool(processes=2) as pool:
    result = dataset.map(
        map_to_pred, batched=True, batch_size=2, fn_kwargs={"pool": pool}
    )
```

> ## Resources

Reviewer: Remove all the resources except for links to **external** ones (ie, not guides or scripts you can find in the docs)

- A blog post on how to deploy Wav2Vec2 for [Automatic Speech Recognition with Hugging Face's Transformers & Amazon SageMaker](https://www.philschmid.de/automatic-speech-recognition-sagemaker).

> ---

Reviewer: You don't need to change any of the API references
